doc/Developing.md

```diff
@@ -177,59 +218,101 @@ From the package:
 
 We can start the application from the [examples/browser](../examples/browser) directory with:
 
-    yarn run start
+```sh
+yarn run start
+```
 
 This command starts the backend application listening on port `3000`. The frontend application should be available on http://localhost:3000.
 
 If you rebuild native Node.js packages for Electron then rollback these changes
 before starting the browser example by running from the root directory:
 
-    yarn run rebuild:browser
+```
+yarn run rebuild:browser
+```
 
 ## Run the Electron-based example application
 
 From the root directory run:
 
-    yarn run rebuild:electron
+```sh
+yarn run rebuild:electron
+```
 
 This command rebuilds native Node.js packages against the version of Node.js
 used by Electron.
 
 It can also be started from the [examples/electron](../examples/electron) directory with:
 
-    yarn run start
+```sh
+yarn run start
+```
 
 ## Rebuilding
 
 In the root directory run:
 
-    yarn run build
+```sh
+yarn run build
+```
 
 ## Watching
 
+### Watch the TypeScript packages
```

Comment:
Then there's nothing to watch?

Do we have such non-TypeScript package?

Reply:
This mostly refers to TypeScript compilation which happens before bundling, so if we have a pure JS extension then there will be nothing to watch.

Reply:
I understand `TypeScript package` as a package written using TypeScript and requiring transpilation.

Reply:
So it's really "typescript files", because some packages may container both *.js and *.ts files, right? Also, not sure how that works, exactly, but don't we have to reload the code when *.js files change, as well?

Reply:
In Theia the watch scripts are related to code transformations. There's basically two steps: TypeScript compilation and Webpack bundling. This documentation refers to the former. Javascript sources don't require any change during the first step as well, but bundling still depend on them indeed.